flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/fs/ForStFlinkFileSystem.java

```diff
@@ -101,6 +142,12 @@ public boolean rename(Path src, Path dst) throws IOException {
         // The rename is not atomic for ForSt. Some FileSystems e.g. HDFS, OSS does not allow a
         // renaming if the target already exists. So, we delete the target before attempting the
         // rename.
+        if (isLocal(src) && isLocal(dst)) {
+            return localFS.rename(src, dst);
+        } else if (isLocal(src) || isLocal(dst)) {
+            throw new IOException("Rename between local and remote is not supported.");
+        }
+
```

Comment:
Should `isLocal` consider the `fileFilter`?

Reply:
Yes, fixed.